src/lib/salad/rope.h

```diff
@@ -440,74 +456,92 @@ rope_extract(struct rope *rope, rope_size_t offset)
 }
 
 /**
- * Erase a single element from the rope. This is a straightforward
- * implementation for a single-element deletion from a rope. A
- * generic cut from a rope involves 2 tree splits and one merge.
+ * Erase `*p_size` elements from the rope if it fits into single node.
+ * Otherwise erase as many elements as fits into the node which holds data
+ * at given offset. Upon return `p_size` is decreased to the number of
+ * deleted elements.
  *
- * When deleting a single element, 3 cases are possible:
- * - offset falls at a node with a single element. In this case we
- *   perform a normal AVL tree delete.
- * - offset falls at the end or the beginning of an existing node
- *   with leaf_size > 1. In that case we trim the existing node
- *   and return.
- * - offset falls inside an existing node. In that case we split
+ * When deleting, 3 cases are possible:
+ * - offset is at the beginning of the node and number of deleted elements
+ *   is more then or equal to the number elements in the node. In this
+ *   case delete the entire node.
+ * - deleted elements reside at the beginning or at the end of the node.
+ *   In that case trim the existing node and return.
+ * - deleted elements reside inside the node. In that case split
  *   the existing node at offset, and insert the tail.
  *
  * The implementation is a copycat of rope_insert(). If you're
  * trying to understand the code, it's recommended to start from
  * rope_insert().
+ *
+ * Return:
+ *  0 - success
+ * -1 - OOM
```

Comment:
Comments like these are not helpful and tend to outdate very quickly. Lets just drop them.

Reply:
You mean the implementation part? I would keep the first paragraph and the return block - it is good to have comment what function do in general.

Implementation part is also useful IMHO. We can move it inside function.

Reply:
I guess I'll keep then. The thing is the API do not check operation validity (assertions at the beginning of `rope_erase`) so the only possible error is OOM. Looks like it good to know it. In particular it is used in the next patch in assertions for `xrow_update_rope_*` return value.

Regularly I don't specify error reasons it the comment too.

Reply:
You still need to convert it to doxygen format.